### PR TITLE
 chain/bitcoind_client: extend filterTx with script spend support

### PR DESCRIPTION
**Pull request by wpaulino**
In this PR, we extend the BitcoindClient to properly detect the spend of an arbitrary script on-chain. This is possible by looking at a transaction's input and re-deriving the PkScript, from its signature script/witness, of the output it's attempting to spend. This is now possible due to a new `btcd` version, which includes `ComputePkScript`. Upon detecting the spend, a chain.RelevantTx will be dispatched.

Depends on https://github.com/btcsuite/btcd/pull/1359.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -19,3 +18,5 @@ require (
 	google.golang.org/genproto v0.0.0-20181127195345-31ac5d88444a // indirect
 	google.golang.org/grpc v1.16.0
 )
+
+replace github.com/btcsuite/btcd => github.com/wpaulino/btcd v0.0.0-20181206032332-6efa63bc1c66
```
Comment (Roasbeef) [inline go.mod]: Comment reminder to remove once the other PR goes in.

File: chain/bitcoind_client.go [hunk 2]
```diff
@@ -1205,16 +1205,47 @@ func (c *BitcoindClient) filterTx(tx *wire.MsgTx,
...
+			// Non-standard outputs can be safely skipped.
+			continue
+		}
+		if _, ok := c.watchedAddresses[addr.String()]; ok {
+			isRelevant = true
+			break
+		}
+	}
+
+	// We'll also cycle through its outputs to determine if it pays to
```
Comment (Roasbeef) [inline chain/bitcoind_client.go]: Similarly, can't we skip this loop if `isRelevant` is set above?

Comment (wpaulino) [reply chain/bitcoind_client.go]: Nope, as `NotifyReceived` for outpoints will also track them for spends, so we still want that logic to happen for backwards compatibility.

**Comment by cfromknecht** [comment]
Do we have any tests that can exercise this new functionality?

**Comment by wpaulino** [comment]
Yeah within `chainntnfs` in `lnd`. Agree that `btcwallet` in general is lacking in tests.

**Pull request closed without merging**